src/Show.js

```diff
@@ -3,13 +3,41 @@ import "./Show.css";
 
 const tumblr_re = /^\d+?\.media\.tumblr\.com$/;
 
+const getGfycatUrl = urlPathname => {
+  const gfycatId = urlPathname.substr(1);
+  const storageKey = "gfycatUrls";
+
+  return new Promise((resolve, reject) => {
+    let gfycatUrls;
+    try {
+      gfycatUrls = JSON.parse(sessionStorage.getItem(storageKey) || "{}");
+    } catch (e) {
+      gfycatUrls = {};
+    }
+    const entry = gfycatUrls[gfycatId];
+    if (entry && entry.expiresAt > new Date()) {
+      resolve(entry.url);
+      return;
+    }
```

Comment:
if an entry is expired, it could be beneficial to remove it from the object so that expired objects aren't serialized.